### PR TITLE
feat(widget-frame-edit-modal): add contents & bind save event

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [ ] `Error / Warning / Lint / Type`

### Description

https://user-images.githubusercontent.com/50662170/210703534-3242f720-0388-4f63-b8b3-1107dfdf2303.mov

- Refactor widget-form store
- Add edit function to widgetFrame

### CODE REVIEW

File: src/services/dashboards/widgets/_components/WidgetFrame.vue [hunk 1]
```diff
@@ -71,7 +71,23 @@
             </div>
         </div>
         <delete-modal :visible.sync="state.visibleDeleteModal" />
-        <p-button-modal :visible.sync="state.visibleEditModal" />
+        <!--sul-lang-->
```
Comment (song2kim) [inline src/services/dashboards/widgets/_components/WidgetFrame.vue:74]: oh?! `sul-lang` lol

File: src/services/dashboards/dashboard-customize/stores/widget-form.ts [hunk 2]
```diff
@@ -67,11 +72,16 @@ export const useWidgetFormStore = defineStore('widget-form', () => {
         state.inheritOptions = inheritOptions;
     };
 
+    const initWidgetForm = (widgetKey: string) => {
+        state.widgetInfo = dashboardDetailInfoStore.dashboardWidgetInfoList.find((w) => w.widgetKey === widgetKey);
+    };
+
     return {
-        ...toRefs(state),
+        state,
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/stores/widget-form.ts:80]: praise: thanks!

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 3]
```diff
@@ -203,6 +203,18 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
         state.widgetDataMap = widgetDataMap;
     };
 
+    const updateWidgetInfo = (widgetKey: string, data: any) => {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:206]: nitpick: 

```suggestion
    const updateWidgetInfo = (widgetKey: string, data: Partial<DashboardContainerWidgetInfo>) => {
```

File: src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts [hunk 4]
```diff
@@ -203,6 +203,18 @@ export const useDashboardDetailInfoStore = defineStore('dashboard-detail-info',
...
+        const targetIndex = state.dashboardWidgetInfoList.findIndex((info) => info.widgetKey === widgetKey);
+        if (targetIndex > -1) {
+            const _dashboardWidgetInfoList = [...state.dashboardWidgetInfoList];
```
Comment (yuda110) [inline src/services/dashboards/dashboard-detail/store/dashboard-detail-info.ts:209]: nitpick: It's safer to use `cloneDeep`.

```suggestion
            const _dashboardWidgetInfoList = cloneDeep(state.dashboardWidgetInfoList);
```

**Review by yuda110 — APPROVED** [review]
LGTM!!!

**Pull request merged**